File: day09/sol2.py
line = input()
numbers = []
sum = []
current = 0
num = 32321523
ans1 = 0
ans2 = 0
done = False

while line != "":
    if not done:
        thenum = int(line)
        numbers.append(thenum)
        current += thenum
        sum.append(current)
        for i in range(0, len(sum) - 2):
            if current - sum[i] == num:
                done = True
                index1 = i + 1
                index2 = len(sum)
                break
    line = input()

newlist = sorted(numbers[index1:index2])
print(newlist)
smallest = newlist[0]
largest = newlist[len(newlist) - 1]
print(smallest + largest)

File: day09/sol2.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef vector<int> vi;
typedef vector<long long> vll;
typedef pair<int,int> pi;
typedef vector<pi> vpi;

#define F first
#define S second
#define PB push_back
#define MP make_pair
#define REP(i,a,b) for (int i = a; i <= b; i++)
#define TRAV(a, x) for (auto& a : x)

void setIO(string name) {
    ios_base::sync_with_stdio(0); cin.tie(0);
    freopen((name+".in").c_str(),"r",stdin);
    freopen((name+".out").c_str(),"w",stdout);
}

void setIO() {
    ios_base::sync_with_stdio(0); cin.tie(0);
}

int main() {
    setIO();

    string line;
    vector<unsigned ll> numbers;
    vector<unsigned ll> sum;
    unsigned long long current = 0;
    unsigned long long num = 32321523;
    unsigned long long ans = 0;
    bool done = false;
    while (getline(cin, line) && line != "") {
      if (!done) {
        unsigned long long thenum = stoull(line);
        numbers.PB(thenum);
        sum.PB(current + thenum);
        current += thenum;
        REP(i, 0, sum.size() - 2) {
          if (current - sum[i] == num) {
            done = true;
            ans = numbers[i] + thenum;
            break;
          }
        }
      }
    }

    cout << ans;
}
